### ISSUE TITLE
setup:upgrade want's to have full var/ writeable

### ISSUE THREAD

**Issue by amenk** [magento/magento2#2406]
When there is a file like var/.hiawatha which is not writable, bin/magento setup:upgrade fails

Error:

```
 Missing write permissions to the following directories: '/WebServer/websites/example/staging/var'  

Exception trace:
 () at /WebServer/websites/www.example.de/staging/setup/src/Magento/Setup/Model/Installer.php:438
 Magento\Setup\Model\Installer->checkInstallationFilePermissions() at /WebServer/websites/www.example.de/staging/setup/src/Magento/Setup/Model/Installer.php:744
 Magento\Setup\Model\Installer->installDataFixtures() at /WebServer/websites/www.example.de/staging/setup/src/Magento/Setup/Console/Command/UpgradeCommand.php:72
 Magento\Setup\Console\Command\UpgradeCommand->execute() at /WebServer/websites/www.example.de/staging/vendor/symfony/console/Symfony/Component/Console/Command/Command.php:257
 Symfony\Component\Console\Command\Command->run() at /WebServer/websites/www.example.de/staging/vendor/symfony/console/Symfony/Component/Console/Application.php:874
 Symfony\Component\Console\Application->doRunCommand() at /WebServer/websites/www.example.de/staging/vendor/symfony/console/Symfony/Component/Console/Application.php:195
 Symfony\Component\Console\Application->doRun() at /WebServer/websites/www.example.de/staging/vendor/magento/framework/Console/Cli.php:49
 Magento\Framework\Console\Cli->doRun() at /WebServer/websites/www.example.de/staging/vendor/symfony/console/Symfony/Component/Console/Application.php:126
 Symfony\Component\Console\Application->run() at /WebServer/websites/www.example.de/staging/bin/magento:25
```

Also this is hard to debug - would be good, if it shows where the error comes from (it is `\Magento\Setup\Model\FilePermissions::checkRecursiveDirectories`)

**Comment by olivermontes** [comment]
more info about this error:
- this user find file to comment 2 lines: $this->checkInstallationFilePermissions();
- comment this lines, can continue process bin/magento setup:upgrade
- link http://magento.stackexchange.com/a/92370/19

**Comment by mazhalai** [comment]
This has been possibly been fixed in MAGETWO-47297. Please standby while we merge this change.

**Comment by chong1227** [comment]
can i fix it of my owl??

**Comment by chong1227** [comment]
can i fix it of my own??

**Comment by piotrekkaminski** [comment]
This was fixed back in Jan and will be part of 2.1. @chong1227 this change was part of much larger change, so it's hard to extract.
